packages/Button/src/Button.styles.js

```diff
@@ -33,19 +27,48 @@ const borderHoverColors = {
   [types.LINK]: "transparent",
 };
 
-const borderStyles = ({ kind }) => css`
-  border-color: ${borderColors[kind]};
+const disabledBorderColors = {
+  [types.DEFAULT]: tokens.border.color,
+  [types.PRIMARY]: tokens.border.color,
+  [types.SECONDARY]: tokens.border.color,
+  [types.DESTRUCTIVE]: tokens.border.color,
+  [types.FLAT]: tokens.border.color,
+  [types.MINOR]: "transparent",
+  [types.LINK]: "transparent",
+};
 
-  &:hover {
-    ${enabled(css`
-      border-color: ${borderHoverColors[kind]};
-    `)}
-  }
+const borderStyles = ({ kind, isDisabled, isActive }) => css`
+  ${!isDisabled &&
+    !isActive &&
+    css`
+      border-color: ${borderColors[kind]};
+      &:hover {
+        border-color: ${borderHoverColors[kind]};
+      }
+    `}
 `;
 
 // States
 
-const disabledStyles = css`
+const focusStyles = css`
+  &:focus {
+    outline: none;
+  }
+
+  html:not([data-whatinput="mouse"]) &:focus,
+  &[data-has-forced-focus]:focus {
+    ${stylers.focusRing.bordered()}
+  }
```

Comment:
This is the refactored block of styling that seems to work fine in both `v4` and `v5`.